clusters/build-clusters/multi01/multi-arch-builder-controller/deploy.yaml

```diff
@@ -0,0 +1,39 @@
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: multi-arch-builder-controller
+  namespace: ci
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      app: multi-arch-builder-controller
+  template:
+    metadata:
+      labels:
+        app: multi-arch-builder-controller
+    spec:
+      serviceAccountName: multi-arch-builder-controller
+      nodeSelector:
+        kubernetes.io/arch: amd64
+      containers:
+      - name: multi-arch-builder-controller
+        image: multi-arch-builder-controller:latest
+        args:
+        - --dry-run=false
+        imagePullPolicy: Always
+        resources:
+          limits:
+            cpu: 100m
+            memory: 128Mi
+          requests:
+            cpu: 100m
+            memory: 128Mi
+        volumeMounts:
+        - name: docker-config
+          mountPath: /.docker
+          readOnly: true
+      volumes:
+      - name: docker-config
+        secret:
+          secretName: multi01-registry-creds
```

Comment:
@hongkailiu I will need some help creating this. I am not sure how the ci-secret-generator will manage it.

Reply:
Yea it should be. I tried to generate it via the secret generator as well.

Reply:
This is an example.


https://github.com/openshift/release/blob/ba1030b7e8acf8f0c6f7274e2132940d6eeb4973/core-services/ci-secret-generator/_config.yaml#L84

The way of forming the docker.config file

https://github.com/openshift/release/blob/ba1030b7e8acf8f0c6f7274e2132940d6eeb4973/core-services/ci-secret-bootstrap/_config.yaml#L4959-L4986

Reply:
I already did. Can you check https://github.com/openshift/release/pull/42551/files#diff-6f809450f5216bc90d0c08b723c9fe080da1358283bbf47c42f05bfc589c49fdR5009-R5019 and https://github.com/openshift/release/pull/42551/files#diff-a1afb2e1637a7b7fd823c19e62f444a3cf739420a7b4a5d2483bfd089977f9efR245-R258

Reply:
Let me try to be more specific. This secret should hold a file `config.json`, technically the `.dockerconfigjson` with the multi01 registry that contains the token of the `multi-arch-builder-controller` service account. Since that token is being rotated, I need to maintain that secret and update it automatically using the `ci-secret-generator`.

I made a try (check my previous comment), but it seems that I am missing an initial configuration which I don't know how to create.